### PR TITLE
Updates ci to utilize tardigrade-ci

### PR DESCRIPTION
**Pull request by userhas404d**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,163 +1,13 @@
...
-	cd tests && go build ./...
-	cd tests && go mod tidy
-
-terratest/test: | guard/program/go
-	cd tests && go test -timeout 20m
-
-test: terratest/test
-
-clean:
+clean::
```
Comment (confusdcodr) [inline Makefile]: What's with the double `:`

Comment (userhas404d) [reply Makefile]: another gnu make quirk. error generated without the double `::`
`Makefile:5: *** target file 'clean' has both : and :: entries.  Stop.`

Comment (lorengordon) [reply Makefile]: https://www.gnu.org/software/make/manual/html_node/Double_002dColon.html

Comment (userhas404d) [reply Makefile]: from my understanding after reading that doc using `<target>::` is treading into "unintended actions" territory. updated to use `<target>/%:` format

Comment (lorengordon) [reply Makefile]: It just means you have more than one `clean` target, and they should both execute independently when calling `make clean`.

File: Makefile [hunk 2]
```diff
@@ -1,163 +1,13 @@
...
-	cd tests && go build ./...
-	cd tests && go mod tidy
-
-terratest/test: | guard/program/go
-	cd tests && go test -timeout 20m
-
-test: terratest/test
-
-clean:
+clean/%:
```
Comment (lorengordon) [inline Makefile]: This is not really the same thing... `make clean/foo` will now delete the vendor/ directory? That doesn't really make sense...

File: Makefile [hunk 3]
```diff
@@ -1,163 +1,13 @@
...
 	rm -rf vendor
 
-VENDOR ?= vendor/github.com/awslabs/aws-config-rules
-vendor: $(VENDOR)
+## Clones the awslabs aws-config-rules repo
+vendor: VENDOR_TARGET ?= github.com/awslabs/aws-config-rules
```
Comment (lorengordon) [inline Makefile]: This is also not quite the same. Not really seeing any reason to change it, based on the rest of this pr?

The prior approach was intended to allow defining multiple vendored repos in `VENDOR`, all of which would be updated by simply calling `make clean && make vendor`. Now you have to call `make vendor VENDOR_TARGET=...` for each vendored repo.

e.g.

```
make clean && make vendor VENDOR="vendor/github.com/awslabs/aws-config-rules vendor/github.com/plus3it/some-custom-config-rules" 
```

File: Makefile (hunk 2 above)
Comment (userhas404d) [reply Makefile]: rgr, reverted the changes to the clean and vendor targets so things should work as expected now

**Pull request merged**